pkg/gateway/server/dispatcher/dispatcher.go

```diff
@@ -358,25 +366,41 @@ func (d *Dispatcher) startAuthProvider(ctx context.Context, namespace, authProvi
 	return url.Parse(strings.TrimSpace(result.Output))
 }
 
-func (d *Dispatcher) ListConfiguredAuthProviders(ctx context.Context, namespace string) ([]string, error) {
+func (d *Dispatcher) ListConfiguredAuthProviders(namespace string) []string {
+	// For now, the only supported namespace for auth providers is the default namespace.
+	if namespace != system.DefaultNamespace {
+		return nil
+	}
+
+	d.configuredAuthProvidersLock.RLock()
+	defer d.configuredAuthProvidersLock.RUnlock()
+
+	return d.configuredAuthProviders
+}
+
+func (d *Dispatcher) UpdateConfiguredAuthProviders() {
+	d.configuredAuthProvidersLock.Lock()
+	defer d.configuredAuthProvidersLock.Unlock()
+
 	var authProviders v1.ToolReferenceList
-	if err := d.client.List(ctx, &authProviders, &kclient.ListOptions{
-		Namespace: namespace,
+	if err := d.client.List(context.Background(), &authProviders, &kclient.ListOptions{
```

Comment:
I think this function should still take a `context.Context`. We could always pass `context.Background()` to it when we need to.